File: burger-club/burgur/src/main/java/restaurante/example/burgur/Entities/Producto.java
package restaurante.example.burgur.Entities;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Producto {
    private Integer id;
    private String nombre;
    private double precio;
    private String descripcion;
    private String imgURL;
    private boolean activo;
    private String categoria;
    private List<String> ingredientes;
    private boolean nuevo;
    private Integer stock; // Nueva propiedad para el stock
    private boolean popular; // Nueva propiedad para productos populares
    
    // Constructor adicional sin stock ni popular (para compatibilidad)
    public Producto(Integer id, String nombre, double precio, String descripcion, 
                   String imgURL, boolean activo, String categoria, 
                   List<String> ingredientes, boolean nuevo) {
        this.id = id;
        this.nombre = nombre;
        this.precio = precio;
        this.descripcion = descripcion;
        this.imgURL = imgURL;
        this.activo = activo;
        this.categoria = categoria;
        this.ingredientes = ingredientes;
        this.nuevo = nuevo;
        this.stock = generarStockAleatorio(); // Stock aleatorio por defecto
        this.popular = false; // No popular por defecto
    }
    
    // Método auxiliar para generar stock aleatorio
    private Integer generarStockAleatorio() {
        return (int) (Math.random() * 50) + 1; // Entre 1 y 50
    }
    
    // Métodos de conveniencia para el estado del stock
    public boolean isStockBajo() {
        return stock != null && stock < 5;
    }
    
    public boolean isStockMedio() {
        return stock != null && stock >= 5 && stock < 15;
    }
    
    public boolean isStockAlto() {
        return stock != null && stock >= 15;
    }
    
    // Método para obtener la clase CSS del stock
    public String getStockCssClass() {
        if (isStockBajo()) return "stock-low";
        if (isStockMedio()) return "stock-medium";
        return "stock-high";
    }
    
    // Método para obtener el estado del producto como texto
    public String getEstadoTexto() {
        if (nuevo) return "NUEVO";
        if (popular) return "POPULAR";
        return "ACTIVO";
    }
    
    // Método para obtener la clase CSS del estado
    public String getEstadoCssClass() {
        if (nuevo) return "status--nuevo";
        if (popular) return "status--popular";
        return "status--activo";
    }
}

File: Burgur/src/app/Model/Producto/producto.ts
import { Adicional } from "../Adicional/adicional";

export class Producto {
    id!: number;          
    nombre!: string;
    descripcion!: string;
    precio!: number;
    categoria!: string;
    imgURL!: string;
    stock!: number;     
    nuevo!: boolean;
    popular!: boolean;
    activo!: boolean;
    // Ingredientes del Producto
    ingredientes: string[] = [];
    // Adicionales de cada Producto
    //? pq puede que no tenga adicionales
    adicionalesPermitidos?: Adicional[];

    // Constructor con Paráemetros sin id
    constructor(nombre: string, descripcion: string, precio: number, categoria: string, imgURL: string, stock: number, nuevo: boolean, popular: boolean, activo: boolean, ingredientes?: string[]) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.categoria = categoria;
        this.imgURL = imgURL;
        this.stock = stock;
        this.nuevo = nuevo;
        this.popular = popular;
        this.activo = activo;
        // Si no se pasan ingredientes, inicializar con array vacío
        this.ingredientes = ingredientes ?? [];
    }
}